src/libostree/ostree-sysroot.c

```diff
@@ -1896,6 +1896,7 @@ ostree_sysroot_simple_write_deployment (OstreeSysroot      *sysroot,
   /* tracks when we come across the booted deployment */
   gboolean before_booted = TRUE;
   gboolean before_merge = TRUE;
+  g_assert (deployments);
```

Comment:
I think in this case the compiler is complaining about the `NULL` that can be returned by the `g_return_val_if_fail()` here:
 https://github.com/ostreedev/ostree/blob/9cdd98125063734ad0c016d51f51a0c8e724645a/src/libostree/ostree-sysroot.c#L1303
Let's directly change that into an assertion maybe?

Reply:
Ahh, nicely spotted!  I changed that and a few others.